service/storage_service.cc

```diff
@@ -4921,6 +4926,18 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(shar
            // Return an error since the command is from outdated leader
            co_return result;
         }
+
+        {
+            auto& state = _raft_topology_cmd_handler_state;
+            if (state.term != term) {
+                state.term = term;
```

Comment:
@gusev-p How do you know if this isn't the old term? In which case, reordering will not be detected.

Reply:
Because earlier we have:

```
            // do barrier to make sure we always see the latest topology
            co_await raft_server.read_barrier(&_abort_source);
            if (raft_server.get_current_term() != term) {
               // Return an error since the command is from outdated leader
               co_return result;
            }
```

Reply:
Right